racetrack_commons/racetrack_commons/api/asgi/asgi_server.py

```diff
@@ -26,20 +28,44 @@
     'GET /metrics/ 200',
 }
 
-UVICORN_DEBUG_LOGS = False
+UVICORN_ERROR_LOGS = True
 
 
 def serve_asgi_app(
     app: Union[ASGIApp, str],
     http_port: int,
     http_addr: str = '0.0.0.0',
     access_log: bool = False,
+    on_shutdown: Optional[Callable[[], None]] = None
 ):
     use_reloader = is_deployment_local() and isinstance(app, str)
     mode_info = ' in RELOAD mode' if use_reloader else ''
     logger.info(f'Running ASGI server on http://{http_addr}:{http_port}{mode_info}')
     _setup_uvicorn_logs(access_log)
-    uvicorn.run(app=app, host=http_addr, port=http_port, log_level="debug", reload=use_reloader)
+
+    config = uvicorn.Config(
+        app=app,
+        host=http_addr,
+        port=http_port,
+        log_level="debug",
+        reload=use_reloader,
+        timeout_graceful_shutdown=3,
+    )
+    server = ManagableServer(config)
+
+    def signal_handler(sig, frame):
```

Comment:
It might be prudent to have the signal handler be a bit more extendable if it isn't too bothersome. Maybe someone wants `SIGHUP` support or something someday. For now the signal handler seems to assume all signals want to terminate it.

There's a decent argument that we are not going to need it though, so I trust your judgement here.

Reply:
I see, maybe I'll just call it `shutdown_signal_handler` since it's only handling SIGINT and SIGTERM right now.